### PR TITLE
Make ChatApp Scalable

### PR DESCRIPTION
**Pull request by sonmn**
• Team3a commit to finally fix chat app by making it scalable. The app now only load a specific room at startup and on room click. 
• Previously the app load all messages for each room, and load all messages * the number of rooms on startup. 

Contributor: Ashu, Son, Michael.

### CODE REVIEW

**Comment by ashsingh21** [comment]
LGTM

**Comment by logydoghan** [comment]
Look good to me; the idea on loading messages upon clicking on each room is restful.

File: group1/communication/node/main.js [hunk 1]
```diff
@@ -182,6 +187,35 @@ var rooms = {
...
+                                'user_id':  userroom_data.user,
+                        },
+                        headers: { 'Content-Type': 'application/json' }
+                };
+                console.log("data room: %s data user: %s", userroom_data.room, userroom_data.user);
+                client.post('http://localhost:8000/api/roomuser/', userroom_template, function(data,response) {
+                        console.log( util.format('(%s) Room user %s updated to "%s"!!! %s', response.statusCode, data.id, data.room_id, data.user_id) );
+                });
+    }
+}*/
```
Comment (oliver-nowak) [inline group1/communication/node/main.js]: @sonmn we should delete commented out code.

**Comment by CS673S17-Team-3** [comment]
LGTM

**Comment by oliver-nowak** [comment]
@sonmn @ashsingh21 @logydoghan can you resolve these merge conflicts? once resolved, i can merge & deploy your code. :)

**Pull request merged**